packages/ember-metal/lib/meta.js

```diff
@@ -104,11 +117,70 @@ Meta.prototype.isInitialized = function(obj) {
   return this.proto !== obj;
 };
 
+const NODE_STACK = [];
+
+Meta.prototype.destroy = function() {
+  // remove chainWatchers to remove circular references that would prevent GC
```

Comment:
destroy should be idempotent, should we not check the meta destroyed flag here and set it?